src/game/Utils/Timer_Control.cc

```diff
@@ -73,52 +73,17 @@ extern UINT32 guiPotCharPathBaseTime;
 
 static UINT32 TimeProc(UINT32 const interval, void*)
 {
+	static TIMECOUNTER lastUpdate;
+
+	auto const now{ std::chrono::steady_clock::now() };
+
 	if (!gfPauseClock)
 	{
-		guiBaseJA2Clock += BASETIMESLICE;
-
-		for (UINT32 i = 0; i != NUMTIMERS; i++)
-		{
-			UPDATECOUNTER(i);
-		}
-
-		// Update some specialized countdown timers...
-		UPDATETIMECOUNTER(giTimerTeamTurnUpdate);
-
-		if (gpCustomizableTimerCallback)
-		{
-			UPDATETIMECOUNTER(giTimerCustomizable);
-		}
-
-#ifndef BOUNDS_CHECKER
-		if (fInMapMode)
-		{
-			// IN Mapscreen, loop through player's team
-			FOR_EACH_IN_TEAM(s, OUR_TEAM)
-			{
-				UPDATETIMECOUNTER(s->PortraitFlashCounter);
-				UPDATETIMECOUNTER(s->PanelAnimateCounter);
-			}
-		}
-		else
-		{
-			// Set update flags for soldiers
-			FOR_EACH_MERC(i)
-			{
-				SOLDIERTYPE* const s = *i;
-				UPDATETIMECOUNTER(s->UpdateCounter);
-				UPDATETIMECOUNTER(s->DamageCounter);
-				UPDATETIMECOUNTER(s->BlinkSelCounter);
-				UPDATETIMECOUNTER(s->PortraitFlashCounter);
-				UPDATETIMECOUNTER(s->AICounter);
-				UPDATETIMECOUNTER(s->FadeCounter);
-				UPDATETIMECOUNTER(s->NextTileCounter);
-				UPDATETIMECOUNTER(s->PanelAnimateCounter);
-			}
-		}
-#endif
+		auto const timeDiff = now - lastUpdate;
+		guiBaseJA2Clock += std::chrono::duration_cast<std::chrono::milliseconds>(timeDiff).count();
 	}
 
+	lastUpdate = now;
```

Comment:
Why not keep this inside the above block? Now when the player unpauses the timers will update as if there was no pause in the interim.

Reply:
I'll rewrite this a little to make the logic how JA2Clock is handled while time is paused easier to follow. The problems with paused time also shouldn't be overemphasized; it happens quite rarely: while either a message box, the item pickup menu or the door open menu is open. This means it's not really an issue for timing parts of the UI, even though such parts could and probably should better use the GetClock() function defined in Timer.h

Reply:
Most of the interval constants are for the gui, so that's good to hear. What I was worried about is quests, bleeding, daytime, that sort of thing. Consider you've got the open door pop-up and leave to make some tea. Can someone die due to that?

Reply:
> Most of the interval constants are for the gui, so that's good to hear. What I was worried about is quests, bleeding, daytime, that sort of thing. Consider you've got the open door pop-up and leave to make some tea. Can someone die due to that?

No, that's one of those things pausing time actually prevents. If you've a wounded merc who would usually bleed every couple of seconds in realtime, he'll not bleed while those menus are open, but instead resume bleeding a couple of seconds after the menu are closed. This behavior is not changed by this commit. 

Quests do not use the JA2Clock() or `TIMECOUNTERs` but the the "game clock" function in GameClock.h, which has it's own functions to interrupt or pause time, so they're not affected either.